### PR TITLE
feat: Add support for kover code coverage reporting

### PR DESCRIPTION
**Pull request by arudge**
Screen Shot 2021-12-06 at 3 27 12 PM

Due to how the test project is setup, I didn't see a better way to make it into a spock `@Unroll`able test.
I ended up duplicating the tests for kover support.

### CODE REVIEW

**Comment by arudge** [comment]
A few items I wasn't sure about....
* Needs some updates to the doc site repo.
* Did you want to bump the version in `:publishers:gradle-plugin-build.gradle` file?

File: publishers/gradle-plugin/src/test/groovy/projektor/plugin/testkit/coverage/KoverCoverageMultiTaskSpec.groovy [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+class KoverCoverageMultiTaskSpec extends SingleProjectSpec {
+
+    @Override
+    boolean includeKoverPlugin() {
+        return true
+    }
+
+    @See("https://github.com/Kotlin/kotlinx-kover - Minimal supported Gradle version: 6.4")
+    @Unroll
+    def "should publish coverage results from unit and integration tests to server with Gradle version #gradleVersion"() {
```
Comment (craigatk) [inline publishers/gradle-plugin/src/test/groovy/projektor/plugin/testkit/coverage/KoverCoverageMultiTaskSpec.groovy:27]: Great test case that covers a project with both unit and integration test sets!

**Review by craigatk — APPROVED** [review]
@arudge thanks so much for adding support for Kover! And great test cases added as part of this change :)

**Comment by craigatk** [comment]
Good call out about the doc site and updating the version in the Gradle build file - I'll take care of those.

**Pull request merged**